internal/counter.go

```diff
@@ -93,12 +82,60 @@ func (t *Counter) topAsSortedList() []*KeyCount {
 }
 
 func (t *Counter) GetTop() []*KeyCount {
-	t.lock.Lock()
-	defer t.lock.Unlock()
 	topList := t.topAsSortedList()
 	if len(topList) > t.size {
 		return topList[0:t.size]
 	} else {
 		return topList
 	}
 }
+
+// merge applies the counts from the SegmentCounter into the Counter.
+// Once merged, the SegmentCounter should be discarded.
```

Comment:
Why discard?  This is end of run, whole process is about to exit, right?

Reply:
right, it's a comment for the future, if someone thinks about changing stuff around. I can take it out if you think its confusing.